src/libraries/Microsoft.PowerFx.Core/Texl/Intellisense/IntellisenseHelper.cs

```diff
@@ -304,15 +304,16 @@ internal static void AddSuggestionsForNamespace(IntellisenseData.IntellisenseDat
         /// <summary>
         /// Adds suggestions that start with the MatchingString from the given type.
         /// </summary>
-        internal static void AddTopLevelSuggestions(IntellisenseData.IntellisenseData intellisenseData, DType type, string prefix = "")
+        internal static void AddTopLevelSuggestions(IntellisenseData.IntellisenseData intellisenseData, DType type, string prefix = "", string filter = null)
         {
             Contracts.AssertValue(intellisenseData);
             Contracts.Assert(type.IsValid);
             Contracts.Assert(prefix.Length == 0 || (TexlLexer.PunctuatorBang + TexlLexer.PunctuatorDot).IndexOf(prefix[prefix.Length - 1]) >= 0);
 
             foreach (var tName in type.GetRootFieldNames().Select(name => (Type: type.GetType(name), Name: name)))
             {
-                if (!intellisenseData.TryAddCustomColumnTypeSuggestions(tName.Type))
+                bool filtered = string.IsNullOrEmpty(filter) ? false : !tName.Name.Value.StartsWith(filter, StringComparison.OrdinalIgnoreCase);
```

Comment:
>bool filtered = string.IsNullOrEmpty(filter) ? false : !tName.Name.Value.StartsWith(filter, StringComparison.OrdinalIgnoreCase); [](http://example.com/codeflow?start=16&length=128)

Will this also prevent us from suggesting substring matches? is that intended? 
ie. If I type `cat`, should I not get `Concat` in suggestion?

Reply:
Removed